code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/ContentCardEventObserver.kt

```diff
@@ -0,0 +1,77 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.messaging
+
+import com.adobe.marketing.mobile.aepcomposeui.AepUI
+import com.adobe.marketing.mobile.aepcomposeui.interactions.UIEvent
+import com.adobe.marketing.mobile.aepcomposeui.observers.AepUIEventObserver
+import com.adobe.marketing.mobile.aepcomposeui.uimodels.SmallImageTemplate
+
+class ContentCardEventObserver(private var callback: ContentCardCallback?) : AepUIEventObserver {
+    companion object {
+        private const val SELF_TAG = "ContentCardEventObserver"
+    }
+
+    override fun onEvent(event: UIEvent<*, *>) {
+        when (event) {
+            is UIEvent.Display -> {
+                when (val template = event.aepUi.getTemplate()) {
+                    is SmallImageTemplate -> {
+                        val eventHandler = SmallImageTemplateEventHandler(callback)
```

Comment:
nit: Do we need to create a new instance of `SmallImageTemplateEventHandler` for each event? Do you think we can create one when the `ContentCardEventObserver` is initialized?

Reply:
the aepUI parameter is only available for the Display, Interact, and Dismiss events. i could cast the aepUI parameter outside of the `when` block but i would still need to determine the type before casting.

Reply:
Ah ok, understood. Do you think we can create the event handler for all types at class initialization and use the appropriate one here after the check? Something like
```

private lateinit val smallImageEventHandler
init {
    SmallImageTemplateEventHandler(callback)
    LargeImageTemplateEventHandler(callback)
    .....
}


override fun onEvent(event: UIEvent<*, *>) {
        when (event) {
            is UIEvent.Display -> {
                when (val template = event.aepUi.getTemplate()) {
                   is SmallImageTemplate -> {
                        val eventHandler = smallImageEventHandler.onEvent(event, template.id)
                    }
                }
            }
       }
}```